crates/feldera-storage/src/file/writer.rs

```diff
@@ -0,0 +1,1031 @@
+        let writer = Self {
+            writer,
+            cws,
+            finished_columns,
+        };
+        Ok(writer)
+    }
+
+    pub fn write<K, A>(&mut self, column: usize, item: (&K, &A)) -> IoResult<()>
+    where
+        K: Rkyv,
+        A: Rkyv,
+    {
+        let row_group = if column + 1 < self.n_columns() {
+            let row_group = self.cws[column + 1].take_rows();
+            assert!(!row_group.is_empty());
+            Some(row_group)
+        } else {
+            None
+        };
+
+        // Add `value` to row group for column.
+        self.cws[column].rows.end += 1;
+        self.cws[column].add_item(&mut self.writer, item, &row_group)
+    }
+
+    pub fn finish_column<K, A>(&mut self, column: usize) -> IoResult<()>
+    where
+        K: Rkyv,
+        A: Rkyv,
+    {
+        debug_assert_eq!(column, self.finished_columns.len());
+        for cw in self.cws.iter().skip(1) {
+            assert!(cw.rows.is_empty());
+        }
+
+        self.finished_columns
+            .push(self.cws[column].finish::<W, K, A>(&mut self.writer)?);
+        Ok(())
+    }
+
+    pub fn close(mut self) -> IoResult<W> {
+        debug_assert_eq!(self.cws.len(), self.finished_columns.len());
+
+        // Write the file trailer block.
+        let file_trailer = FileTrailer {
+            header: BlockHeader::new(b"LFFT"),
+            version: VERSION_NUMBER,
+            columns: take(&mut self.finished_columns),
+        };
+        write_block(&mut self.writer, &mut file_trailer.into_block())?;
+
+        Ok(self.writer)
+    }
+
+    pub fn n_columns(&self) -> usize {
+        self.cws.len()
+    }
+
+    pub fn n_rows(&self) -> u64 {
+        self.cws[0].rows.end
+    }
+}
+
+/// 1-column layer file writer.
+///
+/// `Writer1<W, K0, A0>` writes a new 1-column layer file in which column 0 has
+/// key and auxiliary data types `(K0, A0)`.
+///
+/// # Example
+///
+/// The following code writes 1000 rows in column 0 with values `(0, ())`
+/// through `(999, ())`.
+///
+/// ```
+/// # use feldera_storage::file::writer::{Parameters, Writer1};
+/// # use std::fs::File;
+/// let mut file = File::create("file.layer").unwrap();
+/// let mut file = Writer1::new(file, Parameters::default()).unwrap();
+/// for i in 0..1000_u32 {
+///     file.write0((&i, &())).unwrap();
+/// }
+/// file.close().unwrap();
+/// ```
+pub struct Writer1<W, K0, A0>
+where
+    W: Write + Seek,
+    K0: Rkyv,
+    A0: Rkyv,
+{
+    inner: Writer<W>,
+    _phantom: PhantomData<(K0, A0)>,
+}
+
+impl<W, K0, A0> Writer1<W, K0, A0>
+where
+    W: Write + Seek,
+    K0: Rkyv,
+    A0: Rkyv,
+{
+    /// Creates a new writer with the given parameters.
+    pub fn new(writer: W, parameters: Parameters) -> IoResult<Self> {
+        Ok(Self {
+            inner: Writer::new(writer, parameters, 1)?,
+            _phantom: PhantomData,
+        })
+    }
+    /// Writes `item` to column 0.  `item.0` must be greater than passed in the
```

Comment:
greater? K0 does not implement PartialOrd.

Reply:
I'll add an `Ord` bound.